### ISSUE TITLE
Problems minifying with Codekit

### ISSUE THREAD

**Issue by dlewand691** [jlmakes/scrollreveal.js#253]
May just be my error for using too many plugins, but when trying to minify and create a source map, I get the error of `"Cannot set property 'ScrollReveal' of undefined"` and page will not load. 

When not minifying and just combining imports, it runs fine.

Site is here: [http://www.paulrand.design/](http://www.paulrand.design/)

_Using Codekit on OSX to process all files._

![screen shot 2016-04-28 at 11 04 59 am](https://cloud.githubusercontent.com/assets/1159913/14890862/2d4e766e-0d32-11e6-810a-cd9f723f25ae.png)

**Comment by jlmakes** [comment]
Hey @dlewand691, not sure what’s going on.

So in either a Node process or the browser, the top-level `this` should be defined—either as `module.exports` or `window` (respectively) but it looks it’s `undefined` for some reason.

#209 brought up a similar issue related to webpack compatibility, and required a configuration change to the build process—maybe there’s still room for improvement.

Out of curiosity @dlewand691, have you narrowed down which build features cause the problem? (For example, is always when you minify or when you minify with source maps?) Plus, can you list some of the other JavaScript libraries you’re using? I would be curious to see how their dist files differ.

**Comment by dlewand691** [comment]
Hi Julian-
Wow, great info. A little more about my set up:
- Tried compiling with a variety of options: JSHint with minify only, then minify + source map, JSLint with minify only, then +source map. All had the same result. 
- Only option that worked was to do no minification
- Use quite a few plugins, eventhough I know I probably shouldn't. Here's a list
  - Foundation framework
  - Jquery
  - Animsition
  - Chocolat
  - Fastclick
  - Unveil
  - matchHeight
  - Owl Carousel
  - RRSSB
  - Scrollup
  - Select2
  - Selection-sharer
  - TRMix

You can also go to [http://www.paulrand.design/js/min/app-min.js](http://www.paulrand.design/js/min/app-min.js) to see the full un-minified JS file in question. I'd definitely like to trim down but may not, will see. 

Thanks for looking into this! Much appreciated. Let me know if you need anymore info.
Danny

**Comment by jlmakes** [comment]
Would you do me a favor @dlewand691?

Remove [line 38](https://github.com/jlmakes/scrollreveal.js/blob/master/dist/scrollreveal.js#L38) and see if it fixes this problem. I’m suspicious that `use strict` is the culprit.

**Comment by dlewand691** [comment]
Sorry, same results.

**Comment by jlmakes** [comment]
@dlewand691 Of course it’s not a one line fix, that’d be too easy 😂

I’d really like to fix this problem, but I’m not sure what to do since I don’t have/use Codekit. I’ve simplified the module scopes to try and wrap my mind around this…

```js

(function(root, factory) {
  if (typeof define === 'function' && define.amd) {
    define(factory);
  } else if (typeof exports === 'object') {
    module.exports = factory(require, exports, module);
  } else {
    root.ScrollReveal = factory();
  }
}(this, function (require, exports, module) {

  ;(function() {
        
    this.ScrollReveal = (function () {})();

  }).call(this);

  return this.ScrollReveal;

}));
```

I’m curious how Codekit is handling it’s minification, because some aggressive settings are too destructive. What’s odd for me is that in my all tests, `this` either references the `window` object (in browsers) or the `global` object (in node).

I haven’t a clue right now what else could be going on.

**Comment by dlewand691** [comment]
In Codekit there are many options that you can enable/disable for either JSHint or JSLint. I'm using the defaults, but can certainly try combinations of them to see if one works.

Some info about is here: [https://incident57.com/codekit/help.html#jslint](https://incident57.com/codekit/help.html#jslint)

And a screenshot of a few of the options (there are many more):
<img width="1012" alt="screen shot 2016-05-27 at 5 37 49 pm" src="https://cloud.githubusercontent.com/assets/1159913/15621836/c16d2596-2431-11e6-97ac-550c2c98c67b.png">

**Comment by jlmakes** [comment]
@dlewand691 Are you using Codekit 1, or the Codekit 2 beta?

**Comment by jlmakes** [comment]
Hey @dlewand691, any chance you can help me out? I’m refactoring some stuff for version 3.2, and I think I may have sorted out this Codekit issue.

If I send you a sample build, can you drop it into your Codekit workflow to see if the problem has gone away?

**Comment by dlewand691** [comment]
Absolutely! Thrilled to help. Send me any files and I'll let you know how things work out. Send to dlewandowski38@yahoo.com

**Comment by vaughn-taylor** [comment]
I'm using CodeKit 3.2, Build 25849 and I'm having a problem minifying as well. Un-minified, I have no JS errors. As soon as I minify, I get the following error:

`Uncaught ReferenceError: c is not defined`

I realize that without looking at my code, it's impossible to say what is going on here, but I was hoping that there has been some sort of resolution and/or tips on how to resolve this issue?

**Comment by jlmakes** [comment]
This issue `Cannot set property 'ScrollReveal' of undefined` was solved in `3.2.0` by removing the UMD gulp plugin, and manually preparing the UMD.

You’re getting a different error @vaughn-taylor. What version of ScrollReveal are you using, and what is the offending line number? Also, have you reproduced this issue with just ScrollReveal and Codekit alone (no other libraries/plugins)?
